code/src/main/kotlin/com/expediagroup/sdk/v2/core/authentication/strategy/BearerAuthenticationHandlerFactory.kt

```diff
@@ -13,14 +14,25 @@ import com.expediagroup.sdk.v2.core.trait.configuration.SecretTrait
 import com.google.api.client.auth.oauth2.ClientCredentialsTokenRequest
 import com.google.api.client.http.BasicAuthentication
 import com.google.api.client.http.GenericUrl
+import com.google.api.client.http.HttpRequestInitializer
 import com.google.api.client.http.HttpTransport
 import com.google.api.client.json.gson.GsonFactory
 import com.google.auth.oauth2.AccessToken
 import java.time.Instant
 import java.util.*
 
-object BearerAuthenticationHandlerFactory: CreateAuthenticationHandlerTrait {
+/**
+ * Factory object for creating Bearer Authentication handlers.
+ */
+object BearerAuthenticationHandlerFactory: AuthenticationHandlerTrait {
 
+    /**
+     * Creates an authentication handler to refresh access tokens using client credentials.
+     *
+     * @param config The client configuration that must implement KeyTrait, SecretTrait, and AuthEndpointTrait.
+     * @param transport The HTTP transport to be used for token refresh requests.
+     * @return An instance of RefreshAccessTokenTrait which can be used to refresh access tokens.
+     */
```

Comment:
I know this PR is to refactor the logging mainly, but since it touches this file, can we consider some refactors?

1. Do you think it's a good idea to group bearer-related traits in a common interface `interface BearerAuthConfig : KeyTrait, SecretTrait, AuthEndpointTrait`? Potentially reducing the number of `require` statements.

2. I feel we are overusing the scoping functions here? can we break down the statements for a clearer flow? 
```kotlin
val tokenRequest = ClientCredentialsTokenRequest(
    transport,
    GsonFactory.getDefaultInstance(),
    GenericUrl(authEndpoint)
).setClientAuthentication(BasicAuthentication(key, secret))

// Attach default initializer
tokenRequest.requestInitializer = extendRequestInitializer(tokenRequest.requestInitializer)

// Log token renewal in progress
logger.info(LoggingMessage.TOKEN_RENEWAL_IN_PROGRESS, LogMessageTag.PROGRESSING)

// Execute the request and handle the response
return try {
    val response = tokenRequest.execute()
    // Build and return the AccessToken
} catch (e: Exception) {
    // Handle exceptions
}

private fun extendRequestInitializer(initializer: HttpRequestInitializer?): HttpRequestInitializer {
  return when (initializer) {
      is ChainedHttpRequestInitializer ->
          initializer.extend(ChainedHttpRequestInitializer.default())
      is HttpRequestInitializer ->
          ChainedHttpRequestInitializer.default().extend(initializer)
      else -> ChainedHttpRequestInitializer.default()
  }
}
``` 

3. Can we use `Instant` instead of `Date`? 
    - `val expirationTime = Instant.now().plusSeconds(response.expiresInSeconds.toLong())`
    
4. I don't see any synchronization logic on `refreshAccessToken`, is this safe?

Reply:
1. I disagree, while it is true that we have multiple requires, but it makes it clear what traits we are expecting in the same place. Also, I would like to push against having a hierarchy of traits, and would like to keep it as flat as possible. It is true that this can be a "only one extra level" thing, but still, I prefer to stick extremely to the rule.
2. Done.
3. No, the builder only accepts `Date` objects.
4. It is handled by the Google client itself. I and @mohnoor94 tested that long time ago.